src/rubrix/monitoring/model_monitor.py

```diff
@@ -3,19 +3,31 @@
 from typing import Any
 
 from ._spacy import Language, ner_monitor
-from ._transformers import TextClassificationPipeline, classifier_monitor
+from ._transformers import Pipeline, huggingface_monitor
+from .base import ModelNotSupportedError
 
 _LOGGER = logging.getLogger(__name__)
 
 
 def monitor(task_model: Any, dataset: str, sample_rate: float = 0.3):
-    if isinstance(task_model, Language):
-        return ner_monitor(task_model, dataset=dataset, sample_rate=sample_rate)
-    if isinstance(task_model, TextClassificationPipeline):
-        return classifier_monitor(task_model, dataset=dataset, sample_rate=sample_rate)
+    try:
+        model_monitor = None
+        if isinstance(task_model, Language):
+            model_monitor = ner_monitor(
+                task_model, dataset=dataset, sample_rate=sample_rate
+            )
+        elif isinstance(task_model, Pipeline):
+            model_monitor = huggingface_monitor(
+                task_model, dataset=dataset, sample_rate=sample_rate
+            )
+        if model_monitor is None:
+            raise ModelNotSupportedError()
+        return model_monitor
+    except ModelNotSupportedError:
+        warnings.warn(
+            "The provided task model is not supported by monitoring module. "
+            "Predictions won't be logged into rubrix"
+        )
```

Comment:
I think the try-except block is not necessary here:
```suggestion
    model_monitor = None
    if isinstance(task_model, Language):
        model_monitor = ner_monitor(
            task_model, dataset=dataset, sample_rate=sample_rate
        )
    elif isinstance(task_model, Pipeline):
        model_monitor = huggingface_monitor(
            task_model, dataset=dataset, sample_rate=sample_rate
        )
        
    if model_monitor is None:
        warnings.warn(
            "The provided task model is not supported by monitoring module. "
            "Predictions won't be logged into rubrix"
        )
        return task_model
        
    return model_monitor
```

Reply:
done